src/uu/skill/src/command.rs

```diff
@@ -0,0 +1,162 @@
+}
+
+#[derive(Debug, Clone)]
+pub enum Expr {
+    Terminal(Vec<String>),
+    User(Vec<String>),
+    Pid(Vec<i32>),
+    Command(Vec<String>),
+    Raw(Vec<String>),
+}
+
+impl Settings {
+    pub fn new(args: ArgMatches) -> Self {
+        let mut signal = args.get_one::<String>("signal").unwrap().to_string();
+        if signal.starts_with("-") {
+            signal.remove(0);
+        }
+        let literal: Vec<String> = args
+            .get_many("expression")
+            .unwrap_or_default()
+            .cloned()
+            .collect();
+        let expr = if args.get_flag("command") {
+            Expr::Command(literal)
+        } else if args.get_flag("user") {
+            Expr::User(literal)
+        } else if args.get_flag("pid") {
+            Expr::Pid(literal.iter().map(|s| s.parse::<i32>().unwrap()).collect())
+        } else if args.get_flag("tty") {
+            Expr::Terminal(literal)
+        } else {
+            Expr::Raw(literal)
+        };
+
+        Self {
+            signal,
+            expression: expr,
+            fast: args.get_flag("fast"),
+            interactive: args.get_flag("interactive"),
+            list: args.get_flag("list"),
+            table: args.get_flag("table"),
+            no_action: args.get_flag("no-action"),
+            verbose: args.get_flag("verbose"),
+            warnings: args.get_flag("warnings"),
+        }
+    }
+}
+
+// Pre-parses the command line arguments and returns a vector of OsString
+
+// Mainly used to parse the signal to make sure it is valid
+// and insert the default signal if it's not present
+pub fn parse_command(args: &mut impl uucore::Args) -> Vec<String> {
+    let option_char_set: HashSet<char> = HashSet::from([
+        '-', 'f', 'i', 'l', 'L', 'n', 'v', 'w', 'c', 'p', 't', 'u', 'h', 'V',
+    ]);
+    let option_set: HashSet<&str> = HashSet::from([
+        "--table",
+        "--list",
+        "--no-action",
+        "--verbose",
+        "--warnings",
+        "--interactive",
+        "--fast",
+        "--command",
+        "--user",
+        "--pid",
+        "--tty",
+        "--help",
+        "--version",
+    ]);
+    let args = args
+        .map(|str| str.to_str().unwrap().into())
+        .collect::<Vec<String>>();
+
+    let exprs = |arg: &String| !arg.starts_with('-');
+    let options = |arg: &String| {
+        (arg.starts_with('-') && arg.chars().all(|c| option_char_set.contains(&c)))
+            || (arg.starts_with("--") && option_set.contains(&arg.as_str()))
+    };
+
+    let signals = args
+        .iter()
+        .filter(|arg: &&String| !exprs(arg))
+        .filter(|arg: &&String| !options(arg))
+        .collect::<Vec<&String>>();
+    if signals.len() == 1 {
+        let signal = &signals[0].as_str()[1..]; // Remove the leading '-'
+        if ALL_SIGNALS.contains(&signal) {
+            args.to_vec()
+        } else {
+            eprintln!("Invalid signal: {}", &signal);
+            std::process::exit(2);
```

Comment:
See my previous comment about returning a `Result`. We usually use something like:
```rust
return Err(USimpleError::new(2, format!("Invalid signal: {}", &signal)));
```